src/actions/SelectedLocationsActions.js

```diff
@@ -0,0 +1,75 @@
+import {
+  DELETE_SELECTED_LOCATION,
+  CLEAR_SELECTED_LOCATIONS,
+  UPDATE_SELECTED_LOCATION,
+  ADD_SELECTED_LOCATION
+} from '../actionTypes';
+import { WEATHER_MIN_WAIT_INTERVAL } from '../constants/constants';
+import { weatherAPI } from '../services/dataService';
+import { setIsFetchingInProgress } from './ServerApiActions';
+
+export const addSelectedLocation = locationData => ({
+  type: ADD_SELECTED_LOCATION,
+  payload: locationData
+});
+
+export const updateSelectedLocation = locationData => ({
+  type: UPDATE_SELECTED_LOCATION,
+  payload: locationData
+});
+
+export const deleteSelectedLocation = id => ({
+  type: DELETE_SELECTED_LOCATION,
+  payload: {
+    id
+  }
+});
+
+export const clearSelectedLocations = () => ({
+  type: CLEAR_SELECTED_LOCATIONS
+});
+
+export const putSelectedLocation = locationData => (dispatch, getState) => {
+  const selectedLocations = getState().selectedLocations;
+
+  if (selectedLocations.some(data => data.id === locationData.id)) {
+    dispatch(updateSelectedLocation(locationData));
+  } else {
+    dispatch(addSelectedLocation(locationData));
+  }
+};
+
+export const updateAllSelectedLocationsData = () => async (dispatch, getState) => {
+  const state = getState();
+  if (state.serverApi.isTokenReceived && !state.serverApi.isFetchingInProgress) {
+    try {
+      dispatch(setIsFetchingInProgress(true));
+
+      state.selectedLocations.forEach(async location => {
```

Comment:
feel a bit iffy about this code... meaning awaiting inside the loop. How about doing all together (pseudo code)?

```
const weatherRequests = state.selectedLocations.map(({ id })=> weatherAPI.getCurrentWeather(id));
const locationRequests = state.selectedLocations.map(({ id }) => weatherAPI. getLocationInfo(id));

const locationInfo = await Promise.all[locationRequests];
const weatherInfo = await Promise.all[weatherRequests];

dispatch(putSelectedLocations(...));

```

?

Reply:
You are right. Your approach is more obvious and the info appears on the screen faster.
Refactored.
Thanks.